packages/cli/src/util/env/add-env-record.ts

```diff
@@ -13,26 +12,22 @@ export default async function addEnvRecord(
   projectId: string,
   type: ProjectEnvType,
   key: string,
-  envValue: string,
-  targets: ProjectEnvTarget[]
+  value: string,
+  targets: ProjectEnvTarget[],
+  gitBranch: string
 ): Promise<void> {
   output.debug(
     `Adding ${type} Environment Variable ${key} to ${targets.length} targets`
   );
-
-  let value = envValue;
-
-  if (type === ProjectEnvType.Secret) {
-    const secret = await client.fetch<Secret>(
-      `/v2/now/secrets/${encodeURIComponent(envValue)}`
-    );
-    value = secret.uid;
-  }
-
-  const body = { type, key, value, target: targets };
-
-  const urlProject = `/v6/projects/${projectId}/env`;
-  await client.fetch<ProjectEnvVariableV5>(urlProject, {
+  const body: Omit<ProjectEnvVariable, 'id'> = {
+    type,
+    key,
+    value,
+    target: targets,
+    gitBranch: gitBranch || undefined,
+  };
+  const url = `/v7/projects/${projectId}/env`;
+  await client.fetch<ProjectEnvVariable>(url, {
```

Comment:
I think we don't need the type since we aren't using the return value?

```suggestion
  await client.fetch(url, {
```